packages/node/src/indexer/worker/worker.ts

```diff
@@ -44,6 +46,8 @@ async function initWorker(startHeight: number): Promise<void> {
     const projectService: ProjectService = app.get('IProjectService');
     // Initialise async services, we do this here rather than in factories so we can capture one off events
     await projectService.init(startHeight);
+    const monitorService = app.get(MonitorService);
+    setMonitorService(monitorService);
```

Comment:
I would move this to the monitor service constuructor. `setMonitorService(this)`